lib/lib.go

```diff
@@ -0,0 +1,285 @@
+package lib
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/spf13/viper"
+)
+
+// Podium is a struct that represents a podium API application
+type Podium struct {
+	httpClient        *http.Client
+	Config            *viper.Viper
+	URL               string
+	baseLeaderboard   string
+	localeLeaderboard string
+}
+
+var (
+	client *http.Client
+	once   sync.Once
+)
+
+// Member maps an member identified by their publicID to their score and rank
+type Member struct {
+	PublicID     string
+	Score        int
+	Rank         int
+	previousRank int
```

Comment:
previousRank must be an exported attribute (PreviousRank) for the unmarshal to work

Reply:
That was intentional though. `previousRank` is not set in any of the functions, since the Podium API requires a query string `?prevRank=true` in order to return this info, and none of the functions in this lib call the API passing that. I only left it there because in future we might want to retrieve this info and it would be as simple as exporting the var.

Reply:
Ok!